cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/BrokerCapacity.java

```diff
@@ -17,12 +17,12 @@ public class BrokerCapacity {
 
     private int id;
     private String cpu;
-    private String disk;
+    private Object disk;
```

Comment:
I changed this to an Object type because the disk configuration can either be a String or JsonObject, depending on whether it is a non-JBOD or JBOD configuration is used respectively. I could force this type to JsonObject but makes the code a little messier IMO, let me know if I should do it anyway!

Reply:
I think we had this discussion somewhere already in the previous PR about intra-broker balancing but did not come up with any better idea. So I guess I can live with it.

@ppatierno @tomncooper Any idea how to make this better?

Reply:
Well, if you want to avoid using Object, you could create a new class that wraps the logic for dealing with this. Something like `CruiseControlDiskCapacity` that has 2 constructors or takes an Object and figures it out? Then you would call `CruiseControlDiskCapacity.getCapacity()` to convert to what you need.

Reply:
I have just added a unit test for checking the disk configuration in the same way CC reads it in. It catches the error that was flagged by system tests. It needs some refactoring bur feel free to play with it in the meantime.

Reply:
`CruiseControlDiskCapacity.getCapacity()` could return whatever `JsonObject.put()` requires (or have another method return the json friendly type `CruiseControlDiskCapacity.getJsonObject()`)?

Reply:
Yeah I was tweaking along these lines, the `JsonObject.put()` wants an `Object` in the end.
In the end we are just wrapping for avoiding an `Object disk` declaration but a fancier `CruiseControlDiskCapacity disk`.
I am playing with it, @kyguy let's sync on this.

Reply:
I created this new `CapacityDisk` class https://gist.github.com/ppatierno/0ca502bd307dffef4bb5eac70988de83

That should be used where we are using `Object` and then when writing the JSON we should have something like this:

```java
private JsonObject generateBrokerCapacity(BrokerCapacity brokerCapacity) {
        return new JsonObject()
            .put("brokerId", brokerCapacity.getId())
            .put("capacity", new JsonObject()
                .put("DISK", brokerCapacity.getDisk().getCapacity())
                .put("CPU", brokerCapacity.getCpu())
                .put("NW_IN", brokerCapacity.getInboundNetwork())
                .put("NW_OUT", brokerCapacity.getOutboundNetwork())
            )
            .put("doc", brokerCapacity.getDoc());
    }
```

To highlight the call at `.put("DISK", brokerCapacity.getDisk().getCapacity())`.

@kyguy let's sync offline on this to maybe getting improvements on the class. @tomncooper wdyt?

Reply:
LGTM

Reply:
I have committed Paolo's changes + some refactoring changes, let me know what you think!

Reply:
LGTM! Thanks Kyle!